process/transaction/shardProcess.go

```diff
@@ -311,6 +311,11 @@ func (txProc *txProcessor) processMoveBalance(
 		return err
 	}
 
+	err = txProc.checkDestinationIsPayable(acntDst)
```

Comment:
this is not quite enough - in case of cross shard move balance to a non payable contract - the value should be sent back. in case of intraShard call the transaction has to be treated as invalid transaction, thus consuming gas - putting into invalid blocks.
In case of crossShard non payable balance transfer - you should call processIfTxErrorCrossShard.
In case of intra shard executingFailedTransaction